docs/interpreter/spark.md

```diff
@@ -365,6 +365,51 @@ myScalaDataFrame = DataFrame(z.get("myScalaDataFrame"), sqlContext)
   </div>
 </div>
 
+### Object Interpolation
+Some interpreters can interpolate object values from `z` into the command string by using the 
+`{variable-name}` syntax. The value of any object previously `put` into `z` can be 
+interpolated into a command string by using such a pattern containing the object's name. 
+The following example shows one use of this facility:
+
+####In Scala cell:
+```
+z.put("minAge", 35)
+```
+
+####In later SQL cell:
+```
+%sql select * from members where age >= {minAge}
+```
+
+The interpolation of a `{var-name}` pattern is performed only when `z` contains an object with the specified name.
+But the pattern is left unchanged if the named object does not exist in `z`.
+Further, all `{var-name}` patterns within the command must must be translatable for any interpolation to occur -- 
+translation of only some of the patterns in a command line is never done.
+
+In some situations, it is necessary to use { and } characters in a command without invoking the 
+object interpolation mechanism. For these cases an escaping mechanism is available -- 
+doubled braces {{ and }} should be used. The following example shows the use of {{ and }} for passing a 
+regular expression containing just { and } into the command.
+
+```
+%sql select * from members where name rlike '[aeiou]{{3}}'
+```
+
+To summarize, patterns of the form `{var-name}` within commands will be interpolated only if a predefined 
+object of the specified name exists. Additionally, all such patterns within the command line should also 
+be translatable for any interpolation to occur. Patterns of the form `{{any-text}}` are translated into `{any-text}`. 
+These translations are performed only when all occurrences of `{`, `}`, `{{`, and `}}` in any command string confirm to one of the two 
+forms described above. A command containing `{` and/or `}` characters used in any other way 
+(than `{var-name}` and `{{any-text}}`) is used as-is without any changes. 
+No error is flagged in any case. This behavior is identical to the implementation of a similar feature in 
+Jupyter's shell invocation using the `!` magic command.
+
+This feature is disabled by default, and must be explicitly turned on by setting either the environment variable 
+`ZEPPELIN_INTERPRETER_INTERPOLATION` or the Java property `zeppelin.interpreter.interpolation` to `true`. 
+As usual, if both are defined, the environment variable's setting has priority.
+
+At present only the SQL and Shell interpreters support object interpolation. 
```

Comment:
should this go to a different doc page? Shell is not part of spark, and sql can be other sql interpreters like hive or postgresql?

Reply:
The following changes have been made:

1) added a short description of this feature in the documentation for the shell interpreter.
2) changed `SQL` here to `SparkSQL`, but have not removed the reference to shell (see explanation below) 

The reason why the description of interpolation is here is that it is one of the features of `ZeppelinContext` - which was already described here before I added the interpolation part. Perhaps in a future release we should reorganize this document (spark.md), and remove `ZeppelinContext` into a different place in the documentation.

Reply:
ok